### ISSUE TITLE
(timing diagram) message labels ignore default style

### ISSUE THREAD

**Issue by chrisaga** [plantuml/plantuml#1746]
**Describe the bug**
Message labels have their own font name, style and size.
They don't follow `defaultFontName`, `defaultFontColor` nor `defaultFontSize`.
The only styling they accept is inheritance of message arrow's color.

**To Reproduce**

1. Copy one of the classic timing diagram examples with message arrows and render it
2. See that the message label's font is different from other fonts
3. Try to change default font style such as: 
~~~~
skinparam defaultFontName "Serif"
skinparam defaultFontSize 18
skinparam defaultFontColor Blue
~~~~
4. Some text rendering change in the diagram but message labels don't

**Expected behavior**
Message labels follow default styling.

**Conclusion**
This should be a first step toward targeted styling of message labels which was what I was trying to achieve. I realize that would be a feature request.
IMHO not following default style is a bug. Feel free to qualify this as a feature request too if I am wrong.

**Comment by The-Lum** [comment]
Hi @chrisaga, and all,

Here is a corresponding example:

From:
```puml
@startuml

robust "DNS Resolver" as DNS
robust "Web Browser" as WB
concise "Web User" as WU

@0
WU is Idle
WB is Idle
DNS is Idle

@+100
WU -> WB : URL
WU is Waiting
WB is Processing

@+200
WB is Waiting
WB -> DNS@+50 : Resolve URL

@+100
DNS is Processing

@+300
DNS is Idle

@WU
@200 <-> @+150 : {150 ms}
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/NP1F2y8m3CNl-HJ1wn2cuaL4Il4YYCYazBm_HGhR2asd1_6xcyuwqLD3Njxl9HKKIX-6hWNmhXeeu6n_kc1XoBKtusTO4d9Z4hMfK7ftfwHf2RNhQqlchQb9K00Y1wtG4XwQre2MKnaZFZM8R96FW_CTut23gZWcdovjiFqrMS_UrOOeDjYqZ2RvDyK8PejidJCd7J7oKaeA_YMjlaBQY9SNpCSjyzWuqXxnwUW9mlGDVzWB)](https://www.planttext.com/?text=NP1F2y8m3CNl-HJ1wn2cuaL4Il4YYCYazBm_HGhR2asd1_6xcyuwqLD3Njxl9HKKIX-6hWNmhXeeu6n_kc1XoBKtusTO4d9Z4hMfK7ftfwHf2RNhQqlchQb9K00Y1wtG4XwQre2MKnaZFZM8R96FW_CTut23gZWcdovjiFqrMS_UrOOeDjYqZ2RvDyK8PejidJCd7J7oKaeA_YMjlaBQY9SNpCSjyzWuqXxnwUW9mlGDVzWB) 

To:
```puml
@startuml

skinparam defaultFontName "SansSerif"
skinparam defaultFontSize 10
skinparam defaultFontColor green

robust "DNS Resolver" as DNS
robust "Web Browser" as WB
concise "Web User" as WU

@0
WU is Idle
WB is Idle
DNS is Idle

@+100
WU -> WB : URL
WU is Waiting
WB is Processing

@+200
WB is Waiting
WB -> DNS@+50 : Resolve URL

@+100
DNS is Processing

@+300
DNS is Idle

@WU
@200 <-> @+150 : {150 ms}
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/RP51QyCm38Nl-XKYrp98EtOPOvYi50gZZ8RWivkenSonX-MiqB7_FYNpCZPwij3J-vw45ISJqz0x07wr_ir4qsD7HpEuj0a-xKnFMDJ6SqtH7elhKxMz42xBw-9JS27YAH9vW1WE0oSirhiQzyJ1lLCiq31AOnOr7R2AuSnPqnMqmRUMwLjhPg41K2Ne1YtZjdC4kfhBCUAd1hLObjFWpQFWy1wR_NFsQMEJzQTiVOcX9UQn8QRLQAh-JGb2s6fnLmed7p7nSaeE_akw_HNoHhAy4Zu-24-C4-rZV7h-14M-ar_v0W00)](https://www.planttext.com/?text=RP51QyCm38Nl-XKYrp98EtOPOvYi50gZZ8RWivkenSonX-MiqB7_FYNpCZPwij3J-vw45ISJqz0x07wr_ir4qsD7HpEuj0a-xKnFMDJ6SqtH7elhKxMz42xBw-9JS27YAH9vW1WE0oSirhiQzyJ1lLCiq31AOnOr7R2AuSnPqnMqmRUMwLjhPg41K2Ne1YtZjdC4kfhBCUAd1hLObjFWpQFWy1wR_NFsQMEJzQTiVOcX9UQn8QRLQAh-JGb2s6fnLmed7p7nSaeE_akw_HNoHhAy4Zu-24-C4-rZV7h-14M-ar_v0W00) 

Regards,
Th.

**Comment by The-Lum** [comment]
Hi @chrisaga, and all,

Thanks for the report,

Yes, it is a bug, because arrow's label seems to take the color from `arrow LineColor` and not from `arrow FontColor`...

Here is an example:
```puml
@startuml

<style>
arrow {
	FontName SansSerif
	FontSize 11
	FontColor green
	LineColor red
}
</style>

robust "DNS Resolver" as DNS
robust "Web Browser" as WB
concise "Web User" as WU

@0
WU is Idle
WB is Idle
DNS is Idle

@+100
WU -> WB : URL
WU is Waiting
WB is Processing

@+200
WB is Waiting
WB -> DNS@+50 : Resolve URL

@+100
DNS is Processing

@+300
DNS is Idle

@WU
@200 <-> @+150 : {150 ms}
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/NP51QyCm38NFrQyGlPQnP6EN4Oh9Hg5GocW8ERk9MWoE3PQxiPN-zoc9bx6TBElfVK_6YgCEyT9RW8BZfwKrw13y1rvXiV4kxdLFM6d75GLpcdgL-IBCy-dougqFU0v43XOxusXg1EhW1iLzOaBmnmj7NBxkApmGU_jEOOcQKHgpsD0HImddf3KbjDwrXcdIwbce0LG6JOs6STjPWgQSoo7YfmQrohDny6ujE7p6-h1BlaQRQDmvMT-2RubvQ8ZfOJ2L_wO48MorUig4anunyb9A2lvBUlmLqaQol18-5i8JuqYx3aVFDr3aElcDRm00)](https://www.planttext.com/?text=NP51QyCm38NFrQyGlPQnP6EN4Oh9Hg5GocW8ERk9MWoE3PQxiPN-zoc9bx6TBElfVK_6YgCEyT9RW8BZfwKrw13y1rvXiV4kxdLFM6d75GLpcdgL-IBCy-dougqFU0v43XOxusXg1EhW1iLzOaBmnmj7NBxkApmGU_jEOOcQKHgpsD0HImddf3KbjDwrXcdIwbce0LG6JOs6STjPWgQSoo7YfmQrohDny6ujE7p6-h1BlaQRQDmvMT-2RubvQ8ZfOJ2L_wO48MorUig4anunyb9A2lvBUlmLqaQol18-5i8JuqYx3aVFDr3aElcDRm00) 

Regards,
Th.

**Comment by jimnelson372** [comment]
Hello.   I have a fix for this, but I have a question about how to deal with the default skins when an element such as the timingdiagram.arrow needs default skin settings.  To have the current defaults on the message text, it needs the Font settings added in the skins.  Otherwise it will get the default Font settings for the timingdiagram itself, which is the bolded player headings.   This, of course, is in the absence of the the scripts overriding the defaults either by skinparam settings, as in the original script provided by @chrisaga for this issue , or by the arrow style settings in the script provided by @The-Lum.

With my current changes, here is what Chris's script generates, given the skinparam settings to make all fonts SanSarif/green/10.

<img width="485" alt="image" src="https://github.com/plantuml/plantuml/assets/66284321/8584e91f-922b-4aca-821a-fd5400c6e4ef">

And here is what @The-Lum's script generates using the Style/arrow settings to make the arrow line red and the Font SanSarif/green/11 :

<img width="494" alt="image" src="https://github.com/plantuml/plantuml/assets/66284321/81f3aabe-6eab-44db-8f5b-c79b91115ce5">

I've also added 2 extra pixels to the left of the labels, so that they aren't touching the arrows.

Now, if I update my plantuml.skin file settings on the timingDiagram, as follows:

```
timingDiagram {
...
	arrow {
	    FontName Serif
	    FontSize 14
            FontStyle plain
            FontColor darkblue
            ...
	}
..
}
```
And go back to the original script above without any overrides from default, I get what looks just like the current diagram, other than the 2 pixel space I just mentioned so that the message isn't right on the line.

<img width="490" alt="image" src="https://github.com/plantuml/plantuml/assets/66284321/7ece12b5-c845-4d7b-b78d-623b2abfd866">

(And, yes, the hardcoded Font settings for the message was Sarif/plain/14.  I am curious why it is larger than labels such as on the timing constraint arrow, but it currently is.   This fix would be an opportunity to choose a new default size/style.):

However, the .skin files are not all updated with this change, then the message now defaults as I mentioned above, to the player headers font settings.  Here is what it looks like without my plantuml.skin changes:

<img width="490" alt="image" src="https://github.com/plantuml/plantuml/assets/66284321/743fbd07-8927-4dc2-82da-6e2fc51f539a">

It's not too bad, but it is different than the expected defaults.   I've thought about somehow detecting that the .skin file doesn't provide the Font settings on the timingDiagram.arrow, and then hardcoding the original default if no override settings are in the script, but I'm not sure how to do that, nor if I need to.   I'm seeking advice on how new diagram features than require .skin changes have been managed in the past.  

Best regards, 

Jim Nelson

**Comment by arnaudroques** [comment]
> It's not too bad, but it is different than the expected defaults. I've thought about somehow detecting that the .skin file doesn't provide the Font settings on the timingDiagram.arrow, and then hardcoding the original default if no override settings are in the script, but I'm not sure how to do that, nor if I need to. I'm seeking advice on how new diagram features than require .skin changes have been managed in the past.

Many thanks for your investigation!
Let's keep the code as simple as possible: so do not try to detect if the .skin file doesn't provide the Font settings on the timingDiagram.arrow.
Even if it slightly changes the rendering of previous diagrams, I think this is an acceptable change.
I am not even sure that anyone would notice the change :-)

The more important thing is that the next-to-be behavior is more consistent than the actual behavior.

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed on [V1.2024.5](https://github.com/plantuml/plantuml/releases/tag/v1.2024.5).
- Thanks @chrisaga for the report.
- Thanks PlantUML team and @jimnelson372 for your work.

Regards.
